extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -415,13 +414,14 @@ export class AmpAnalytics extends AMP.BaseElement {
     }
     const key = this.expandTemplate_(spec['sampleOn'], trigger);
 
-    return urlReplacementsFor(this.getWin()).expand(key).then(key => {
-      const digest = this.sha384_(key);
-      if (digest[0] % 100 < spec['threshold']) {
-        return resolve;
-      }
-      return Promise.resolve(false);
-    });
+    const keyPromise = urlReplacementsFor(this.getWin()).expand(key);
+    const cryptoPromise = cryptoFor(this.getWin());
+    return Promise.all([keyPromise, cryptoPromise])
+        .then(([key, crypto]) => {
```

Comment:
Yes. Seems working, at least under chrome. Is it something new?
On Jun 29, 2016 5:09 PM, "Malte Ubl" <notifications@github.com> wrote:

> In extensions/amp-analytics/0.1/amp-analytics.js
> <https://github.com/ampproject/amphtml/pull/3841#discussion_r69050419>:
>
> > @@ -415,13 +414,14 @@ export class AmpAnalytics extends AMP.BaseElement {
> >      }
> >      const key = this.expandTemplate_(spec['sampleOn'], trigger);
> >
> > -    return urlReplacementsFor(this.getWin()).expand(key).then(key => {
> > -      const digest = this.sha384_(key);
> > -      if (digest[0] % 100 < spec['threshold']) {
> > -        return resolve;
> > -      }
> > -      return Promise.resolve(false);
> > -    });
> > +    const keyPromise = urlReplacementsFor(this.getWin()).expand(key);
> > +    const cryptoPromise = cryptoFor(this.getWin());
> > +    return Promise.all([keyPromise, cryptoPromise])
> > +        .then(([key, crypto]) => {
>
> Wow, this works?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/3841/files/ad4493954bc4e6803beefa696a9284258ecd4f00#r69050419>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ANd2kIU8IoiJL-4whBCC0Pyo8dEMUOxKks5qQwlMgaJpZM4JBrfz>
> .
>
> --
> You received this message because you are subscribed to the Google Groups
> "amphtml-eng-github" group.
> To unsubscribe from this group and stop receiving emails from it, send an
> email to amphtml-eng-github+unsubscribe@google.com.
> To post to this group, send email to amphtml-eng-github@google.com.
> To view this discussion on the web visit
> https://groups.google.com/a/google.com/d/msgid/amphtml-eng-github/ampproject/amphtml/pull/3841/r69050419%40github.com
> <https://groups.google.com/a/google.com/d/msgid/amphtml-eng-github/ampproject/amphtml/pull/3841/r69050419%40github.com?utm_medium=email&utm_source=footer>
> .
>